apps/mosaico/src/Main.purs

```diff
@@ -385,14 +394,37 @@ notFound env _ _ = do
     appendMosaico mosaicoString env.htmlTemplate >>= appendHead (mkWindowVariables windowVars)
   pure $ Response.notFound $ StringBody $ html
 
+setAuthCookie :: { guards :: { credentials :: Maybe UserAuth } } -> Aff (Response ResponseBody)
+setAuthCookie { guards: { credentials } } =
+  setCredentials credentials <$> pure (Response.ok EmptyBody)
+
+-- Payload 0.4.0 can't do multiple Set-Cookie headers
+setCredentials :: forall a. Maybe UserAuth -> Response a -> Response a
+setCredentials Nothing response = response
+setCredentials (Just { authToken, userId }) (Response response@{ headers }) = Response
+  response { headers = Headers.set "Set-Cookie" cookies headers }
+  where
+    cookies = "CombinedAuth=" <> (\(Token t) -> t) authToken <> "|" <> UUID.toString userId <>
+              "; Secure; SameSite=Strict; HttpOnly; Path=/"
```

Comment:
Should we set a TTL also?

Reply:
I guess. A session cookie wouldn't survive a browser restart. I'll make it a year.